### PR TITLE
Implemented a start screen and refactored the UI components that it uses

### PR DESCRIPTION
**Pull request by mcgill-a**
image

### CODE REVIEW

**Review by athompsonScottLogic — APPROVED** [review]
code reviewed 👏 👏

File: whichone/static/js/buttons.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  $("#mute-icon").on("click", function (event) {
+    if (user.muteSound) {
+      user.muteSound = false;
+      document.getElementById("mute-icon").src =
+        "/static/resources/volume-on.png";
+    } else {
+      user.muteSound = true;
+      document.getElementById("mute-icon").src =
+        "/static/resources/volume-off.png";
+    }
```
Comment (connor-settle) [inline whichone/static/js/buttons.js]: Above if statement is unnecessary, could be simplified

File: whichone/static/js/data.js [hunk 2]
```diff
@@ -96,7 +96,9 @@ async function request(key, callback) {
       user[key] = JSON.parse(data);
       user["expire"] = new Date().getTime();
       localStorage.setItem(spotify_id, JSON.stringify(user));
-      callback();
+      if (callback != null) {
+        callback();
```
Comment (connor-settle) [inline whichone/static/js/data.js]: Might want to consider best way to handle async code. You are using a mixture of promises and callbacks. I think generally promises (& async/await) is preferable.

File: whichone/static/js/game.js [hunk 3]
```diff
@@ -477,3 +342,42 @@ function updateLives() {
...
+    }
+  });
+
+  let mean = total / array.length;
+
+  return {
+    scores: user.scores,
+    min: min,
+    max: max,
+    mean: mean,
```
Comment (connor-settle) [inline whichone/static/js/game.js]: You can use shorthand for this where the local variable has the same name as the object key:
```
return {
  scores: user.scores,
  min,
  max,
  mean
}
```

File: whichone/static/js/game.js [hunk 4]
```diff
@@ -477,3 +342,42 @@ function updateLives() {
...
+  };
+}
+
+function getScoreData() {
+  let data = calculateScoreData(user.scores);
```
Comment (connor-settle) [inline whichone/static/js/game.js]: Could be const

There are a few places you could use const, maybe worth adding to linting rules.

File: whichone/static/js/game.js [hunk 5]
```diff
@@ -477,3 +342,42 @@ function updateLives() {
...
+  data.stdev = getStandardDeviation(data.scores, data.mean);
```
Comment (connor-settle) [inline whichone/static/js/game.js]: Can you just set stdev in the calculateScoreData function?

**Review by connor-settle — COMMENTED** [review]
Looks good. Added a few minor code level comments.

**Comment by mcgill-a** [comment]
> Looks good. Added a few minor code level comments.

@connor-settle I have implemented all of your suggested changes in the latest commit

**Review by connor-settle — APPROVED** [review]

**Pull request merged**